test/Microsoft.AspNetCore.SystemWebAdapters.CoreServices.Tests/CacheTests.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Web;
+using System.Web.Caching;
+using System.Web.SessionState;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.SystemWebAdapters.SessionState;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters
+{
+    public class CacheTests
+    {
+
+        [Fact]
+        public void CacheFromHttpContext()
+        {
+            // Arrange
+            var cache = new Cache();
+
+            var serviceProvider = new Mock<IServiceProvider>();
+            serviceProvider.Setup(s => s.GetService(typeof(Cache))).Returns(cache);
+
+            var httpRuntime = Microsoft.Extensions.DependencyInjection.HttpRuntimeFactory.Create(serviceProvider.Object);
+            HttpRuntime.Current = httpRuntime;
```

Comment:
Any usage of the HttpRuntime outside of statics themselves should be done through DI so we don't need to set this in tests.